users/views.py

```diff
@@ -1,6 +1,17 @@
+from django.conf import settings
+from django.contrib.auth.views import LoginView, LogoutView
+from django.http.response import HttpResponseRedirect
 from django.shortcuts import render  # noqa
-from django.views import generic
 
 
-class LoginView(generic.TemplateView):
+class UserLoginView(LoginView):
     template_name = 'auth/login.html'
+
+    def get(self, request, *args, **kwargs):
+        if request.user.is_authenticated():
+            return HttpResponseRedirect(settings.LOGIN_REDIRECT_URL)
```

Comment:
If you change `LOGIN_REDIRECT_URL = reverse_lazy('battles:list')` to `LOGIN_REDIRECT_URL = 'battles:list'` you won't need to override the get method.

Reply:
Even when I set the paths in settings, I had to use `redirect_authenticated_user = True` in my LoginView to make the redirection, otherwise my authenticated user would still see my login form. Reference: https://docs.djangoproject.com/en/2.0/topics/auth/default/#django.contrib.auth.views.LoginView